selenium/integration/page.level.errors.spec.ts

```diff
@@ -0,0 +1,140 @@
+/*
+ * Copyright © 2022 Cask Data, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the 'License'); you may not
+ * use this file except in compliance with the License. You may obtain a copy of
+ * the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an 'AS IS' BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+
+import { By } from 'selenium-webdriver';
+import { BASE_URL, CDAP_URL, METADATA_URL, PIPELINES_URL } from '../support/constants';
+import { dataTestId, buildChromeDriver, loginIfRequired } from '../support/utils';
+
+let driver;
+const FAKE_NAMESPACE = 'fakeNamespace';
+const SELECTOR_404_MSG = dataTestId('page-404-error-msg');
+const SELECTOR_404_DEFAULT_MSG = dataTestId('page-404-default-msg');
+
+
+describe('Page level error because of ', () => {
+  beforeAll(async () => {
+    driver = buildChromeDriver();
+    await loginIfRequired(driver);
+  });
+
+  it('no namespace in home page should show 404', async () => {
+    // Go to home page
+    await driver.get(`${CDAP_URL}/${FAKE_NAMESPACE}`);
+    await driver.sleep(1000);
```

Comment:
I agree that we could reduce the falkiness but using waits. we have an example [here](https://github.com/cdapio/cdap-ui/blob/a2ce302a55b62a4ba6054b8a32cb187e6da65876/selenium/integration/admin.spec.ts#L59) with the required constants set.

Reply:
You could add this to the utils.ts file
```javascript
export const waitUntilElementLocated = async (driver, cssLocator) => {
  const element = await driver.wait(until.elementLocated(By.css(cssLocator)), TEST_TIMEOUT_TIME, TEST_TIMEOUT_MESSAGE, RETRY_INTERVAL);
  return element;
}
```
and then use it like 
```javascript
const keyEl = await waitUntilElementLocated(driver, addedKeyCssLocator);
```

Reply:
on second thought, this will help with locating elements only (not navigating through urls). feel free to make it more generic or create a separate one for `driver.get` ones. this example from selenium docs might help:
```javascript
const documentInitialised = () =>
    driver.executeScript('return initialised');

await driver.get('file:///race_condition.html');
await driver.wait(() => documentInitialised(), 10000);
const element = driver.findElement(By.css('p'));
assert.strictEqual(await element.getText(), 'Hello from JavaScript!');
```

Reply:
@mrahanjam Thanks! Yeah I'll add both and update related tests